dataset/src/test/scala/frameless/NumericTests.scala

```diff
@@ -98,4 +98,24 @@ class NumericTests extends TypedDatasetSuite {
 
     check(prop _)
   }
+
+  test("modulo") {
```

Comment:
There is a problem with `Float`, `Double` and `BigDecimal`. The compiler cannot find the evidences for those types as Integral. 
`NumericTests.scala:119: could not find implicit value for evidence parameter of type Integral[BigDecimal]`
and actually there exists objects such as `scala.math.Numeric.BigDecimalAsIfIntegral`that is `Integral[BigDecimal]`. The same goes with `Float` and `Double`

Unfortunately importing them directly (`implicit val bd = BigDecimalAsIfIntegral`) leads to compile-time error:
`ambiguous implicit values:
[error]  both method shrinkFractional in object Shrink of type [T](implicit evidence$2: Fractional[T])org.scalacheck.Shrink[T]
[error]  and method shrinkIntegral in object Shrink of type [T](implicit evidence$3: Integral[T])org.scalacheck.Shrink[T]
[error]  match expected type org.scalacheck.Shrink[scala.math.BigDecimal]
`
This is probably why those types are conflicting. Can you think of why does this happen and how to overcome the problem?

Reply:
I'm not sure why it fails, but we really need to use `scala.math.Integral` for this test. I would just make a new local type class for that purpose, `PrimitiveIntegralForTests` or something.